IHostingEnvironment obsolete warnings when building Studio

## Describe the bug
Warnings when building desogner

## To Reproduce
Build Altinn Studio.

```
Configuration/DefaultFileFactory.cs(13,26): warning CS0618: 'IHostingEnvironment' is obsolete: 'This type is obsolete and will be removed in a future version. The recommended alternative is MicrosoftspNetCore.Hosting.IWebHostEnvironment.' [/designer/Designer.csproj]
Configuration/DefaultFileFactory.cs(20,91): warning CS0618: 'IHostingEnvironment' is obsolete: 'This type is obsolete and will be removed in a future version. The recommended alternative is MicrosoftspNetCore.Hosting.IWebHostEnvironment.' [/designer/Designer.csproj]
RepositoryClient/Client/ApiClient.cs(240,34): warning CS0618: 'RestClient.ExecuteTaskAsync(IRestRequest)' is obsolete: 'This method will be renamed to ExecuteAsync soon' [/designer/Designer.csproj]  
    3 Warning(s)
    0 Error(s)
```

## Expected behavior
No warnings.

```
Build succeeded.
    0 Warning(s)
    0 Error(s)
```

## Tasks
- [x] Fix IHostingEnvironment
- [x] Fix RestClient.ExecuteTaskAsync

--- a/src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/AppResourcesSI.cs
+++ b/src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/AppResourcesSI.cs
@@ -22,7 +22,7 @@ namespace Altinn.App.Services.Implementation
     {
         private readonly AppSettings _settings;
         private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly IWebHostEnvironment _hostingEnvironment;
         private readonly ILogger _logger;
 
         private readonly Dictionary<string, string> _assemblyNames = new Dictionary<string, string>();
@@ -36,7 +36,7 @@ namespace Altinn.App.Services.Implementation
         public AppResourcesSI(
             IOptions<AppSettings> settings,
             IHttpContextAccessor httpContextAccessor,
-            IHostingEnvironment hostingEnvironment,
+            IWebHostEnvironment hostingEnvironment,
             ILogger<AppResourcesSI> logger)
         {
             _settings = settings.Value;

--- a/src/studio/src/designer/backend/Configuration/DefaultFileFactory.cs
+++ b/src/studio/src/designer/backend/Configuration/DefaultFileFactory.cs
@@ -10,14 +10,14 @@ namespace Altinn.Studio.Designer.Configuration
     public class DefaultFileFactory : IDefaultFileFactory
     {
         private readonly ServiceRepositorySettings _settings;
-        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly IWebHostEnvironment _hostingEnvironment;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="DefaultFileFactory"/> class.
         /// </summary>
         /// <param name="repositorySettings"> The service repository settings. </param>
         /// <param name="hostingEnvironment"> The hosting environment. </param>
-        public DefaultFileFactory(IOptions<ServiceRepositorySettings> repositorySettings, IHostingEnvironment hostingEnvironment)
+        public DefaultFileFactory(IOptions<ServiceRepositorySettings> repositorySettings, IWebHostEnvironment hostingEnvironment)
         {
             _settings = repositorySettings.Value;
             _hostingEnvironment = hostingEnvironment;

--- a/src/studio/src/designer/backend/Configuration/GeneralSettings.cs
+++ b/src/studio/src/designer/backend/Configuration/GeneralSettings.cs
@@ -70,7 +70,7 @@ namespace Altinn.Studio.Designer.Configuration
         {
             get
             {
-                return "App/Config/Authorization/policy.xml";
+                return "App/config/authorization/policy.xml";
             }
         }
 

--- a/src/studio/src/designer/backend/Controllers/ErrorController.cs
+++ b/src/studio/src/designer/backend/Controllers/ErrorController.cs
@@ -18,7 +18,7 @@ namespace Altinn.Studio.Designer.Controllers
         /// <summary>
         /// Action for local error handling
         /// </summary>
-        /// <param name="webHostEnvironment">IHostingEnvironment</param>
+        /// <param name="webHostEnvironment">IWebHostEnvironment</param>
         /// <returns></returns>
         [Route("/error-local-development")]
         [ApiExplorerSettings(IgnoreApi = true)]
@@ -48,7 +48,7 @@ namespace Altinn.Studio.Designer.Controllers
         /// <summary>
         /// Action for other environment handling than local
         /// </summary>
-        /// <param name="webHostEnvironment">IHostingEnvironment</param>
+        /// <param name="webHostEnvironment">IWebHostEnvironment</param>
         /// <returns></returns>
         [Route("/error")]
         [ApiExplorerSettings(IgnoreApi = true)]

--- a/src/studio/src/designer/backend/Designer.csproj
+++ b/src/studio/src/designer/backend/Designer.csproj
@@ -49,7 +49,7 @@
     <PackageReference Include="Microsoft.DiaSymReader.Native" Version="1.7.0" />
     <PackageReference Include="Markdig" Version="0.16.0" />
     <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="3.1.0" />
-    <PackageReference Include="RestSharp" Version="106.10.1-alpha.0.9" />
+    <PackageReference Include="RestSharp" Version="106.10.1" />
     <PackageReference Include="Serilog.Extensions.Hosting" Version="3.0.0" />
     <PackageReference Include="Serilog.Extensions.Logging" Version="3.0.1" />
     <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />

--- a/src/studio/src/designer/backend/RepositoryClient/Client/ApiClient.cs
+++ b/src/studio/src/designer/backend/RepositoryClient/Client/ApiClient.cs
@@ -237,7 +237,7 @@ namespace Altinn.Studio.Designer.RepositoryClient.Client
                 pathParams,
                 contentType);
             InterceptRequest(request);
-            var response = await RestClient.ExecuteAsync(request, new System.Threading.CancellationToken());
+            var response = await RestClient.ExecuteAsync(request, default(System.Threading.CancellationToken));
             InterceptResponse(request, response);
             return (object)response;
         }

--- a/src/studio/src/designer/backend/RepositoryClient/Client/ApiClient.cs
+++ b/src/studio/src/designer/backend/RepositoryClient/Client/ApiClient.cs
@@ -237,7 +237,7 @@ namespace Altinn.Studio.Designer.RepositoryClient.Client
                 pathParams,
                 contentType);
             InterceptRequest(request);
-            var response = await RestClient.ExecuteTaskAsync(request);
+            var response = await RestClient.ExecuteAsync(request, new System.Threading.CancellationToken());
             InterceptResponse(request, response);
             return (object)response;
         }

